plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,485 @@
+										'Automattic Address Suggestion has been disabled due to invalid JWT token'
+									);
+									return [];
+								case 'rate_limit_exceeded':
+									permanentlyDisabledServices.push( key );
+									setTimeout( () => {
+										permanentlyDisabledServices.splice(
+											permanentlyDisabledServices.indexOf(
+												key
+											),
+											1
+										);
+									}, response.headers.get( 'RateLimit-Retry-After' ) * 1000 );
+									console.error(
+										'Automattic Address Suggestion has been disabled due to rate limit exceeded'
+									);
+									return [];
+								case 'missing_query':
+								case 'no_suggestions':
+									// Cache empty results to avoid repeated API calls
+									cacheResult( inputValue, country, [] );
+									return [];
+								case 'missing_session_id':
+									sessionId =
+										crypto && crypto.randomUUID
+											? crypto.randomUUID()
+											: Math.random()
+													.toString( 36 )
+													.substring( 2 );
+									return [];
+								case 'woo_addression_suggestion_internal_error':
+								case 'woo_address_suggestion_service_error':
+									serviceErrorRetries++;
+									if (
+										serviceErrorRetries >=
+										MAX_SERVICE_ERROR_RETRIES
+									) {
+										permanentlyDisabledServices.push( key );
+										console.error(
+											'Automattic Address Suggestion has been disabled due to internal service error'
+										);
+									}
+									return [];
+								default:
+									return [];
+							}
+						}
+						if ( Array.isArray( data ) ) {
+							data = data.map( ( item ) => ( {
+								id: item.id,
+								label: item.label,
+								matchedSubstrings: item.matched_substrings,
+							} ) );
+							// Cache the successful result
+							cacheResult( inputValue, country, data );
+							return data;
+						}
+					} catch ( e ) {
+						if ( e.name === 'AbortError' ) {
+							// Ignore abort errors from cancelled requests
+							return [];
+						}
+						console.error(
+							'Error fetching address suggestions:',
+							e
+						);
+						return [];
+					}
+				},
+				300,
+				{ leading: false, trailing: true }
+			);
+			window.wc.addressAutocomplete.registerAddressAutocompleteProvider( {
+				id: key,
+				canSearch: () => {
+					try {
+						if ( permanentlyDisabledServices.includes( key ) ) {
+							return false;
+						}
+
+						// Split JWT into parts
+						const [ , payload ] = value.split( '.' );
+						if ( ! payload ) {
+							permanentlyDisabledServices.push( key );
+							return false;
+						}
+
+						// Decode payload
+						const decodedPayload = JSON.parse( atob( payload ) );
+
+						// Check expiration
+						const currentTime = Math.floor( Date.now() / 1000 );
+						if (
+							! decodedPayload.exp ||
+							decodedPayload.exp < currentTime
+						) {
+							permanentlyDisabledServices.push( key );
+							return false;
+						}
+
+						return true;
```

Comment:
Can search validate if the token we have is valid, we don't check country here, maybe we should make this extendible at some point.